examples/fluids/src/setuplibceed.c

```diff
@@ -96,25 +95,21 @@ PetscErrorCode CreateOperatorForDomain(Ceed ceed, DM dm, SimpleBC bc,
   if (phys->has_neumann) {
     // --- Setup
     ierr = DMGetLabel(dm, "Face Sets", &domain_label); CHKERRQ(ierr);
-    ierr = DMGetDimension(dm, &dim); CHKERRQ(ierr);
-    if (dim == 2) num_face = 4;
-    if (dim == 3) num_face = 6;
 
     // --- Get number of quadrature points for the boundaries
     CeedInt num_qpts_sur;
     CeedBasisGetNumQuadraturePoints(ceed_data->basis_q_sur, &num_qpts_sur);
 
-    // --- Create Sub-Operator for each face
-    for (CeedInt i=0; i<num_face; i++) {
+    // --- Create Sub-Operator for inflow boundaries
+    for (CeedInt i=0; i < bc->num_inflow; i++) {
       CeedVector          q_data_sur;
-      CeedOperator        op_setup_sur, op_apply_sur;
-      CeedElemRestriction elem_restr_x_sur, elem_restr_q_sur,
-                          elem_restr_qd_i_sur;
+      CeedOperator        op_setup_sur, op_apply_inflow;
+      CeedElemRestriction elem_restr_x_sur, elem_restr_q_sur, elem_restr_qd_i_sur;
 
       // ---- CEED Restriction
-      ierr = GetRestrictionForDomain(ceed, dm, height, domain_label, i+1,
-                                     Q_sur, q_data_size_sur, &elem_restr_q_sur,
-                                     &elem_restr_x_sur, &elem_restr_qd_i_sur);
+      ierr = GetRestrictionForDomain(ceed, dm, height, domain_label, bc->inflows[i],
```

Comment:
is this change giving the same value?

Reply:
Yes! Just checked it again and it is giving me the correct value for Traveling Vortex.